init.zsh

```diff
@@ -14,6 +14,13 @@ fi
 # Source user configuration
 [[ -s ${ZDOTDIR:-${HOME}}/.zimrc ]] && source ${ZDOTDIR:-${HOME}}/.zimrc
 
+# If necessary, set vi mode before loading modules to ensure history search with arrow keys works in vi mode
+if [ ${zinput_mode}='vi' ]; then
+  set -o vi
+else
+  set -o emacs
+fi
+
```

Comment:
I think this should go to the `input` module.

Reply:
The entire point of the fix is that it has to be set before loading modules. If you could discover why that is the case, we could choose a better place for it, but I don't know enough ZSH to do so at the moment.

Reply:
Agreed. And I think the only place where there should be a documentation about this is here (not in the main README file).